rs/nns/cmc/src/main.rs

```diff
@@ -1569,6 +1569,62 @@ fn memo_to_intent_str(memo: Memo) -> String {
     }
 }
 
+/// Returns Ok if transaction matches expected_memo.
+///
+/// First, looks at the memo field. If that does not match, falls back to the
+/// icrc1_field. If that is of length 8, converts assuming little endian, and if
+/// that matches, returns Ok.
+fn transaction_has_expected_memo(
+    transaction: &Transaction,
+    expected_memo: Memo,
+) -> Result<(), NotifyError> {
+    fn stringify_memo(memo: Memo) -> String {
```

Comment:
Could this be 

```
impl Display for Memo
```
?

NCR; you decide

Reply:
That thought did crossed my mind. Let me [ask FI team][ask], and if they like it, I'll do it in a follow up PR.

[ask]: https://dfinity.slack.com/archives/C01AVALTNSH/p1736256950483319